sbi/simulators/simutils.py

```diff
@@ -518,9 +557,9 @@ def simulate_in_batches(
     n_chunks = math.ceil(num_samples / simulation_batch_size)
     parameter_batches = torch.chunk(parameters, chunks=n_chunks)
 
-    xs = []
-    for batch in parameter_batches:
-        with torch.no_grad():
-            xs.append(simulator(batch))
+    with torch.no_grad():
+        xs = torch.cat([simulator(batch) for batch in parameter_batches])
 
-    return torch.tensor(parameters), torch.cat(xs)
+    # XXX: we need this cast here to avoid
```

Comment:
XXX We construct a tensor here again because a memory-sharing cast
XXX via `as_tensor` results in a RuntimeError: Trying to backward through the graph a second time
XXX when doing multiple rounds in SNPE (gradient-tracking problem to investigate).